### PR TITLE
create dot file

### PR DESCRIPTION
**Pull request by boyingl**
can successful create dot files (default)
to be improved: 
1. some repetition
2. creating onlyblock, onlytype requires compiling with a different option, need to call the compiler multiple times?

### CODE REVIEW

File: CheckerPrinter/src/checkerprinter/InMemory.java [hunk 1]
```diff
@@ -92,11 +92,35 @@ protected boolean initCheckerArgs(JsonObject optionsObject) {
...
+        this.checkerOptionsList = new ArrayList<String>();
+
+        this.checkerOptionsList.add("-Xbootclasspath/p:" + this.CHECKER_FRAMEWORK + "/checker/dist/jdk8.jar");
+        this.checkerOptionsList.add("-processor");
+        this.checkerOptionsList.add(checker);
+
         if (optionsObject.getBoolean("has_cfg")) {
+
+            //ugly, how can this be improved?
+            if(! setupDir(new File("../CFG"))){
```
Comment (CharlesZ-Chen) [inline CheckerPrinter/src/checkerprinter/InMemory.java]: As we discussed before, at current stage you can ignore the multi-user call problem and focus on implementing the main functions. So as a temporary workaround, we can just manually create a `CFG` directory in the server and here just assume there should have a `CFG` directory in the expected location.

File: CheckerPrinter/src/checkerprinter/InMemory.java [hunk 2]
```diff
@@ -153,4 +179,20 @@ protected boolean initCheckerArgs(JsonObject optionsObject) {
             this.checkerPrinter.printDiagnosticReport(diagnosticList);
         }
     } 
+
+    //create a directory, if exists already, delete and recreate a directory
+    protected static boolean setupDir(File dir) {
```
Comment (CharlesZ-Chen) [inline CheckerPrinter/src/checkerprinter/InMemory.java]: Maybe we don't need this method. As we can just assume there will be a `CFG` directory at the moment.

File: CheckerPrinter/src/checkerprinter/InMemory.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ David Pritchard (daveagp@gmail.com), created May 2013
 
     String mainClass;
     String exceptionMsg;
-    List<String> checkerOptionsList;
+    List<String> checkerOptionsList;
```
Comment (wmdietl) [inline CheckerPrinter/src/checkerprinter/InMemory.java]: Avoid adding trailing spaces.

File: CheckerPrinter/src/checkerprinter/InMemory.java [hunk 4]
```diff
@@ -92,13 +92,19 @@ protected boolean initCheckerArgs(JsonObject optionsObject) {
...
+        this.checkerOptionsList.add("-processor");
+        this.checkerOptionsList.add(checker);
+
         if (optionsObject.getBoolean("has_cfg")) {
-            // String cfgLevel = optionsObject.getString("cfg_level");
-            // TODO: add CFG Visualization
+
+            //assume exists CFG directory
+            this.checkerOptionsList.add("-classpath");
+            this.checkerOptionsList.add(this.CHECKER_FRAMEWORK + "/checker/dist/checker.jar");
```
Comment (wmdietl) [inline CheckerPrinter/src/checkerprinter/InMemory.java]: Isn't it a bit surprising that the `checker.jar` file is only needed when that flag is given? You also need it when normally running the processor. If this is indeed needed, can you add a comment documenting why?

**Comment by CharlesZ-Chen** [comment]
Closed as this PR is not ready to merge into `master` branch. New PR is here: #30 .

**Pull request closed without merging**